websites/P/Planet Minecraft/presence.ts

```diff
@@ -0,0 +1,341 @@
+const presence = new Presence({
+		clientId: "1193411747898458132",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	stateless = "stateless";
+
+async function getStrings() {
+	return presence.getStrings({
+		viewHome: "general.viewHome",
+		viewMember: "general.viewMember",
+		viewChannel: "general.viewChannel",
+		readingThread: "general.readingThread",
+		browsing: "general.browsing",
+		reading: "general.reading",
+	});
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		details: "Viewing page:",
+		largeImageKey: "https://i.imgur.com/pNsonk0.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	if (!strings) strings = await getStrings();
+
+	const pathname = document.location.pathname.replaceAll(/\/$/g, "");
+
+	if (pathname === "") {
+		presenceData.state = strings.viewHome;
+		delete presenceData.details;
+	} else {
+		const url = `${document.location.protocol}//${document.location.host}${document.location.pathname}`;
+		if (pathname.startsWith("/member"))
+			memberActivity(presenceData, pathname, url);
+		else if (
+			pathname.startsWith("/account/manage") &&
+			pathname.endsWith("/new")
+		) {
+			presenceData.details = "Managing Submissions";
+			presenceData.state = document.querySelector("#manage_header").textContent;
+		} else {
+			switch (pathname) {
```

Comment:
Why not make this one switch (true)

Reply:
This is a very interested switch usage; I've never seen it before. I would think that my current approach performs better than the one you suggested, but this is going off a limited knowledge of JS under the hood. I am interested to see what [other people have to say](https://stackoverflow.com/q/77787154/5645656).